godot-macros/src/derive_to_variant.rs

```diff
@@ -0,0 +1,231 @@
+                #body
+            }
+        }
+    })
+}
+
+fn make_named_enum_field(named: &venial::NamedStructFields) -> TokenStream {
+    let fields = named.fields.iter().map(|(field, _)| &field.name);
+    quote!(
+        {#(#fields ,)*}
+    )
+}
+
+fn make_tuple_enum_field(s: &venial::TupleStructFields) -> TokenStream {
+    let fields = s
+        .fields
+        .iter()
+        .enumerate()
+        .map(|(k, _)| format_ident!("__{}", k));
+    quote!(
+        (#(#fields ,)*)
+    )
+}
+
+fn make_enum_named_arm(
+    fields: &venial::NamedStructFields,
+    variant_name_string: String,
+) -> TokenStream {
+    let fields = fields
+        .fields
+        .iter()
+        .map(|(field, _)| (field.name.clone(), field.name.to_string()))
+        .map(|(ident, ident_string)| {
+            quote!(
+                root.insert(#ident_string,#ident.to_variant());
+            )
+        });
+    quote! {
+        let mut root = godot::builtin::Dictionary::new();
+        #(
+            #fields
+        )*
+        godot::builtin::dict!{ #variant_name_string : root}.to_variant()
+    }
+}
+
+fn make_enum_tuple_arm(
+    fields: &venial::TupleStructFields,
+    variant_name_string: String,
+) -> TokenStream {
+    if fields.fields.len() == 1 {
+        return quote! {godot::builtin::dict! { #variant_name_string : __0}.to_variant()};
+    }
+    let fields = fields
+        .fields
+        .iter()
+        .enumerate()
+        .map(|(k, _)| format_ident!("__{}", k))
+        .map(|ident| {
+            quote!(
+                root.push(#ident.to_variant());
+            )
+        });
+    quote! {
+        let mut root = godot::builtin::Array::new();
+        #(
+            #fields
+
+        )*
+        godot::builtin::dict!{ #variant_name_string: root }.to_variant()
+    }
+}
+
+fn make_struct_named(
+    body: &mut TokenStream,
+    fields: &venial::NamedStructFields,
+    string_ident: String,
+) {
+    let fields = fields.fields.items().map(|nf| {
+        let field_name = nf.name.clone();
+        let field_name_string = nf.name.to_string();
+
+        quote!(
+            fields.insert(#field_name_string, self.#field_name.to_variant());
+        )
+    });
+
+    *body = quote! {
+        #body
+        let mut fields = godot::builtin::Dictionary::new();
+        #(
+            #fields
+        )*
+        root.insert(#string_ident, fields.to_variant());
+    };
+}
+
+fn make_struct_tuple(
+    body: &mut TokenStream,
+    fields: &venial::TupleStructFields,
+    string_ident: String,
+) {
+    if fields.fields.len() == 1 {
+        *body = quote! {
+            #body
+            root.insert(#string_ident, self.0.to_variant());
+        };
+
+        return;
+    }
+    let fields = fields
+        .fields
+        .iter()
+        .enumerate()
+        .map(|(k, _)| proc_macro2::Literal::usize_unsuffixed(k))
+        .map(|ident| {
+            quote!(
+                fields.push(self.#ident.to_variant());
+            )
```

Comment:
Use `quote! { ... }` syntax.

Please also replace several other occurrences of `quote!( ... )`.